### PR TITLE
feat(syncer): add new sync type for gosec

### LINKED ISSUE #676: Add support for `gosec` as a scanner sync type
Similar to what we've done for `trivy`, `syft`, etc. implement https://github.com/securego/gosec as a sync type (a security checker for Go programs). We should be able to access output as `JSON`.

### PR DESCRIPTION
**Pull request by riyaz-ali**
This PR adds a new sync type for the `gosec` tool. Closes #676

### CODE REVIEW

**Comment by patrickdevivo** [comment]
@riyaz-ali looks great - but I was testing against this repo: https://github.com/mergestat/gosec-issue which I setup to have a `gosec` issue:

```
gosec-issue git:(main) gosec -fmt json .
[gosec] 2023/01/12 23:15:13 Including rules: default
[gosec] 2023/01/12 23:15:13 Excluding rules: default
[gosec] 2023/01/12 23:15:13 Import directory: /Users/patrickdevivo/Desktop/mergestat/gosec-issue
[gosec] 2023/01/12 23:15:13 Checking package: main
[gosec] 2023/01/12 23:15:13 Checking file: /Users/patrickdevivo/Desktop/mergestat/gosec-issue/main.go
{
	"Golang errors": {},
	"Issues": [
		{
			"severity": "MEDIUM",
			"confidence": "HIGH",
			"cwe": {
				"id": "326",
				"url": "https://cwe.mitre.org/data/definitions/326.html"
			},
			"rule_id": "G401",
			"details": "Use of weak cryptographic primitive",
			"file": "/Users/patrickdevivo/Desktop/mergestat/gosec-issue/main.go",
			"code": "5: func main() {\n6: \th := md5.New()\n7: \th.Write([]byte(\"hello\"))\n",
			"line": "6",
			"column": "7",
			"nosec": false,
			"suppressions": null
		},
		{
			"severity": "MEDIUM",
			"confidence": "HIGH",
			"cwe": {
				"id": "327",
				"url": "https://cwe.mitre.org/data/definitions/327.html"
			},
			"rule_id": "G501",
			"details": "Blocklisted import crypto/md5: weak cryptographic primitive",
			"file": "/Users/patrickdevivo/Desktop/mergestat/gosec-issue/main.go",
			"code": "2: \n3: import \"crypto/md5\"\n4: \n",
			"line": "3",
			"column": "8",
			"nosec": false,
			"suppressions": null
		}
	],
	"Stats": {
		"files": 1,
		"lines": 8,
		"nosec": 0,
		"found": 2
	},
	"GosecVersion": "dev"
}%
```

But when I ran the syncer I didn't see any detected issues in the DB:

image

image

So I wonder if something is not getting executed as expected?

File: internal/syncer/gosec_repo_scan.go [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+
+	// indicate that we're starting a gitleaks scan
+	if err := w.sendBatchLogMessages(ctx, []*syncLog{{Type: SyncLogTypeInfo, RepoSyncQueueID: j.ID,
+		Message: fmt.Sprintf(LogFormatStartingSync, j.SyncType, j.Repo),
+	}}); err != nil {
+		return fmt.Errorf("send batch log messages: %w", err)
+	}
+
+	var stdout, stderr bytes.Buffer
+	cmd := exec.CommandContext(ctx, "gosec", "-no-fail", "-fmt", "json", "./...")
```
Comment (patrickdevivo) [inline internal/syncer/gosec_repo_scan.go:69]:
```suggestion
	cmd := exec.CommandContext(ctx, "gosec", "-no-fail", "-fmt", "json", "./...")
```

curious about the `"./..."` arg here - shouldn't it be the root of the cloned repo?

Comment (riyaz-ali) [reply internal/syncer/gosec_repo_scan.go:67]: so, if I run `gosec` w/o any path arguments, it errors saying:

```shell
> gosec -no-fail -fmt json
Error: FILE [FILE...] or './...' or -r expected

# ... usage
```

I tried the `-r` flag but it seems to be not implemented 🙄 So, instead, I added `./...` and set the `cmd.Dir = tmpPath` (the root of the cloned repository)

Comment (riyaz-ali) [reply internal/syncer/gosec_repo_scan.go:67]: we could've done `$tmpPath/...` but I'm not sure how `gosec` would behave (does it treat it as an import?)

**Comment by riyaz-ali** [comment]
@patrickdevivo seems to work for me 🙄 

image

**Review by patrickdevivo — APPROVED** [review]

**Pull request merged**